tests/framework/clients/rancher/catalog/clusterrepo.go

```diff
@@ -0,0 +1,84 @@
+package catalog
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+
+	"github.com/rancher/rancher/pkg/api/steve/catalog/types"
+	scheme "github.com/rancher/rancher/pkg/generated/clientset/versioned/scheme"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+const (
+	rancherChartsURL = "v1/catalog.cattle.io.clusterrepos/rancher-charts"
+	rancherAppsURL   = "v1/catalog.cattle.io.apps/"
+)
+
+// GetLastestChartVersion is used to get the lastest version of `chartName`
+func (c *Client) GetLastestChartVersion(chartName string) (string, error) {
+	result, err := c.RESTClient().Get().
+		AbsPath(rancherChartsURL).Param("link", "index").
+		VersionedParams(&metav1.GetOptions{}, scheme.ParameterCodec).
+		Do(context.TODO()).Raw()
+
+	if err != nil {
+		return "", err
+	}
+
+	var mapResponse map[string]interface{}
+	if err = json.Unmarshal(result, &mapResponse); err != nil {
+		return "", err
+	}
+
+	entries := mapResponse["entries"]
+	specifiedChartEntries := entries.(map[string]interface{})[chartName].([]interface{})
+	if len(specifiedChartEntries) < 1 {
+		return "", fmt.Errorf("failed to find chart %s from the chart repo", chartName)
+	}
+
+	lastestVersion := specifiedChartEntries[0].(map[string]interface{})["version"].(string)
+	return lastestVersion, nil
+}
+
+// ChartInstall installs the chart according to the paratmeter `chart`
+func (c *Client) ChartInstall(chart *types.ChartInstallAction) error {
+	bodyContent, err := json.Marshal(chart)
+	if err != nil {
+		return err
+	}
+
+	result := c.RESTClient().Post().
+		AbsPath(rancherChartsURL).Param("action", "install").
+		VersionedParams(&metav1.CreateOptions{}, scheme.ParameterCodec).
+		Body(bodyContent).
+		Do(context.TODO())
+
+	return result.Error()
+}
+
+// ChartUninstall uninstalls the chart according to `chartNamespace` and `chartName`
+func (c *Client) ChartUninstall(chartName, chartNamespace string) error {
+	uninstallAction := &types.ChartUninstallAction{
```

Comment:
why not expose this object? these could influence the tests.  You could split it into 2 methods if you want to keep a simple option available. `SimpleUninstallChart` and `UninstallChart`

Reply:
I decided to keep just the one `UninstallChart` with the `ChartUninstallAction` exposed.